appengine/requests/src/test/java/com/example/appengine/requests/LoggingServletTest.java

```diff
@@ -0,0 +1,101 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.appengine.requests;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Unit tests for {@link LoggingServlet}.
+ */
+@RunWith(JUnit4.class)
+public class LoggingServletTest {
+  // Set up a helper so that the ApiProxy returns a valid environment for local testing.
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper();
+
+  // To capture and restore stderr
+  private final ByteArrayOutputStream stderr = new ByteArrayOutputStream();
+  private static final PrintStream REAL_ERR = System.err;
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private LoggingServlet servletUnderTest;
+
+  @Before
+  public void setUp() throws Exception {
+    //  Capture stderr to examine messages written to it
+    System.setErr(new PrintStream(stderr));
+
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    servletUnderTest = new LoggingServlet();
+  }
+
+  @After 
+  public void tearDown() {
+    //  Restore stderr  
+    System.setErr(LoggingServletTest.REAL_ERR);
+
+    helper.tearDown();
+  }
+
+  @Test
+  public void testListLogs() throws Exception {
+    servletUnderTest.doGet(mockRequest, mockResponse);
+
+    String out = stderr.toString();
+    
+    // We expect three log messages to be created
+    // with the following messages.
+    assertThat(out).contains("An informational message.");
+    assertThat(out).contains("A warning message.");
+    assertThat(out).contains("An error message.");
+
+    // We expect three log messages to be created
+    // with the following severities.
+    // Since there's no guarantee of case, use lowercase
+    String lcOut = out.toLowerCase();
```

Comment:
Not sure this will find what you are looking for.  "An informational message." contains "info".  "A warning message." contains "warning", etc.